okio/jvm/src/main/java/okio/Pipe.kt

```diff
@@ -116,6 +145,48 @@ class Pipe(internal val maxBufferSize: Long) {
     override fun timeout(): Timeout = timeout
   }
 
+  /**
+   * Writes any buffered contents of this pipe to `sink`, then replace this pipe's source with
+   * `sink`. This pipe's source is closed and attempts to read it will throw an
+   * [IllegalStateException].
+   *
+   * This method must not be called while concurrently accessing this pipe's source. It is safe,
+   * however, to call this while concurrently writing this pipe's sink.
+   */
+  @Throws(IOException::class)
+  fun fold(sink: Sink) {
```

Comment:
What is the expected behavior of folding a pipe a second time?

Reply:
good catch! it should fail with an IllegalStateException